File: Structural/Flyweight/Delphi/Flyweight.pas
unit Flyweight;

interface

uses
  SysUtils, Classes;

type
  TFlyweight = class
  private
    FSharedState: string;
  public
    constructor Create(const SharedState: string);
    procedure Operation(const UniqueState: string);
  end;

implementation

constructor TFlyweight.Create(const SharedState: string);
begin
  inherited Create;
  FSharedState := SharedState;
end;

procedure TFlyweight.Operation(const UniqueState: string);
begin
  Writeln('Operation with shared state:', FSharedState, ' and unique state:', UniqueState);
end;

end.

File: Structural/Flyweight/C#/FlyweightExample/FlyweightExample/Flyweight.cs
namespace FlyweightExample;

public class Flyweight
{
	private readonly string _sharedState;

	public Flyweight(string sharedState)
	{
		_sharedState = sharedState;
	}

	public void Operation(string uniqueState)
	{
		Console.WriteLine($"The operation was performed with the general state '{_sharedState}' and unique condition '{uniqueState}'.");
	}
}